*** matrix_prod.f.orig	2015-11-03 22:39:21.047551929 +0100
--- matrix_prod.f	2015-11-03 22:41:11.867417318 +0100
***************
*** 100,101 ****
--- 100,106 ----
        PARAMETER (IMAG1=(0D0,1D0))
+       INTEGER NAMPSO, NSQAMPSO
+       PARAMETER (NAMPSO=2, NSQAMPSO=3)
+       LOGICAL CHOSEN_SO_CONFIGS(NSQAMPSO)
+       DATA CHOSEN_SO_CONFIGS/.FALSE.,.TRUE.,.FALSE./
+       SAVE CHOSEN_SO_CONFIGS
  C     
***************
*** 108,113 ****
  C     
!       INTEGER I,J
        COMPLEX*16 ZTEMP
        REAL*8 DENOM(NCOLOR), CF(NCOLOR,NCOLOR)
!       COMPLEX*16 AMP(NGRAPHS), JAMP(NCOLOR)
        COMPLEX*16 W(18,NWAVEFUNCS)
--- 113,118 ----
  C     
!       INTEGER I,J,M,N
        COMPLEX*16 ZTEMP
        REAL*8 DENOM(NCOLOR), CF(NCOLOR,NCOLOR)
!       COMPLEX*16 AMP(NGRAPHS), JAMP(NCOLOR,NAMPSO)
        COMPLEX*16 W(18,NWAVEFUNCS)
***************
*** 116,117 ****
--- 121,126 ----
  C     
+ C     FUNCTION
+ C     
+       INTEGER SQSOINDEX
+ C     
  C     GLOBAL VARIABLES
***************
*** 157,164 ****
        MATRIX_PROD = 0.D0
!       DO I = 1, NCOLOR
!         ZTEMP = (0.D0,0.D0)
!         DO J = 1, NCOLOR
!           ZTEMP = ZTEMP + CF(J,I)*JAMP(J)
          ENDDO
-         MATRIX_PROD = MATRIX_PROD+ZTEMP*DCONJG(JAMP(I))/DENOM(I)
        ENDDO
--- 166,179 ----
        MATRIX_PROD = 0.D0
!       DO M = 1, NAMPSO
!         DO I = 1, NCOLOR
!           ZTEMP = (0.D0,0.D0)
!           DO J = 1, NCOLOR
!             ZTEMP = ZTEMP + CF(J,I)*JAMP(J,M)
!           ENDDO
!           DO N = 1, NAMPSO
!             IF (CHOSEN_SO_CONFIGS(SQSOINDEX(M,N))) THEN
!               MATRIX_PROD = MATRIX_PROD + ZTEMP*DCONJG(JAMP(I,N))/DENOM(I)
!             ENDIF
!           ENDDO
          ENDDO
        ENDDO
